src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -2781,21 +2963,43 @@ HRESULT CProfilerManager::InitializeForAttach(
 
     m_bAttach = true;
 
-    vector<CComPtr<CConfigurationSource>> configSources;
+    if (pvClientData == nullptr ||
+        cbClientData == 0)
+    {
+        // No configuration provided
+        return S_FALSE;
```

Comment:
>S_FALSE [](start = 15, length = 7)

Should this be considered a failure? I think if you return S_FALSE, the CLR is still going to initialize CLRIE and keep it attached. Might cause issues down the line if we think that the engine is working properly when we inspect a process that already has a profiler attached but no instrumentation methods.

Reply:
Makes sense. We don't want CLRIE to be attached when it can't support adding additional IM's yet. I'll return E_FAIL.

---
In reply to: [386720873](https://github.com/microsoft/CLRInstrumentationEngine/pull/216#discussion_r386720873) [](ancestors = 386720873)